src/lib.rs

```diff
@@ -510,20 +481,15 @@ impl<T: Translation> Mapping<T> {
     /// the relevant TTBR to a different address.
     ///
     /// It is called automatically by [`deactivate`](Self::deactivate).
-    pub fn mark_inactive(&mut self) {
-        self.previous_ttbr = None;
+    pub fn mark_inactive(&self) {
+        self.active_count.fetch_sub(1, Ordering::SeqCst);
```

Comment:
This will silently wrap if it is called when `active_count` is 0. I think you should check the return value of `fetch_sub` to detect this, and panic if so.